File: services/tradelog/tradelog-api-spec/src/main/java/com/example/tradelog/api/spec/model/Action.java
package com.example.tradelog.api.spec.model;

public enum Action {
    UNKNOWN,
    SELL,
    BUY;

    static public Action lookup(String id) {
        try {
            return Action.valueOf(id);
        } catch (IllegalArgumentException ex) {
            return UNKNOWN;
        }
    }
}

File: services/tradelog/tradelog-api-spec/src/main/kotlin/com/example/tradelog/api/spec/model/Action.kt
package com.example.tradelog.api.spec.model

enum class Action {
    UNKNOWN,
    SELL,
    BUY;

    companion object {
        fun lookup(id: String) : Action {
            return try {
                valueOf(id)
            } catch (ex: IllegalArgumentException) {
                UNKNOWN
            }
        }
    }
}
